Packs/Phishing/Scripts/PhishingDedupPreprocessingRule/PhishingDedupPreprocessingRule.py

```diff
@@ -0,0 +1,209 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+import pandas as pd
+from bs4 import BeautifulSoup
+from sklearn.feature_extraction.text import TfidfVectorizer
+from numpy import dot
+from numpy.linalg import norm
+from email.utils import parseaddr
+import tldextract
+NO_FETCH_EXTRACT = tldextract.TLDExtract(suffix_list_urls=None)
+pd.options.mode.chained_assignment = None  # default='warn'
+
+
+SIMILARITY_THRESHOLD = 0.98
+
+EMAIL_BODY_FIELD = 'emailbody'
+EMAIL_SUBJECT_FIELD = 'emailsubject'
+EMAIL_HTML_FIELD = 'emailbodyhtml'
+FROM_FIELD = 'emailfrom'
+FROM_DOMAIN_FIELD = 'fromdomain'
+MERGED_TEXT_FIELD = 'mereged_text'
+MIN_TEXT_LENGTH = 50
+DEFAULT_ARGS = {
+    'limit': '1000',
+    'incidentTypes': 'Phishing',
+    'exsitingIncidentsLookback': '100 days ago',
+}
+FROM_POLICY_TEXT_ONLY = 'TextOnly'
+FROM_POLICY_EXACT = 'Exact'
+FROM_POLICY_DOMAIN = 'Domain'
+
+FROM_POLICY = FROM_POLICY_TEXT_ONLY
+
+
+def get_existing_incidents(input_args):
+    global DEFAULT_ARGS
+    get_incidents_args = {}
+    for arg in ['incidentTypes', 'query', 'limit']:
+        if arg in input_args:
+            get_incidents_args[arg] = input_args[arg]
+        elif arg in DEFAULT_ARGS:
+            get_incidents_args[arg] = DEFAULT_ARGS[arg]
+    if 'exsitingIncidentsLookback' in input_args:
+        get_incidents_args['fromDate'] = input_args['exsitingIncidentsLookback']
+    elif 'exsitingIncidentsLookback' in DEFAULT_ARGS:
+        get_incidents_args['fromDate'] = DEFAULT_ARGS['exsitingIncidentsLookback']
+    incidents_query_res = demisto.executeCommand('GetIncidentsByQuery', get_incidents_args)
+    if is_error(incidents_query_res):
+        return_error(get_error(incidents_query_res))
+    incidents = json.loads(incidents_query_res[-1]['Contents'])
+    return incidents
+
+
+def extract_domain(address):
+    if address == '':
+        return ''
+    email_address = parseaddr(address)[1]
+    ext = NO_FETCH_EXTRACT(email_address)
```

Comment:
wrap it with a function, function should be lower case, it's weird to have it like "CONST"